src/model.cpp

```diff
@@ -100,10 +100,25 @@ bool Model::doAddComponent(const ComponentPtr &component)
     return ComponentEntity::doAddComponent(component);
 }
 
-void Model::addUnits(const UnitsPtr &units)
+bool Model::addUnits(const UnitsPtr &units)
 {
+    if (units == nullptr) {
+        return false;
+    }
+
+    // Prevent adding multiple times to list.
+    if (hasUnits(units)) {
+        return false;
+    }
+
+    // Prevent adding to multiple models: move units to this model.
+    if (units->hasParent()) {
+        auto otherParent = std::dynamic_pointer_cast<Model>(units->parent());
+        otherParent->removeUnits(units);
+    }
```

Comment:
Again turning an add into a move.

Reply:
That's what addComponent does ... see Andre's comment earlier.

Reply:
I can't see his comment

Reply:
https://github.com/cellml/libcellml/pull/621#discussion_r437822306

Reply:
https://github.com/cellml/libcellml/pull/621#discussion_r437822306